modules/internal/ChapelHashtable.chpl

```diff
@@ -502,31 +491,23 @@ module ChapelHashtable {
     }
 
     proc maybeShrinkAfterRemove() {
-      if (tableNumFullSlots*8 < tableSize && tableSizeNum > 0) {
+      if (tableNumFullSlots*8 < tableSize) {
         resize(grow=false);
       }
     }
 
     // #### rehash / resize helpers ####
 
-    proc _findPrimeSizeIndex(numKeys:int) {
-      //Find the first suitable prime
-      var threshold = (numKeys + 1) * 2;
-      var prime = 0;
-      var primeLoc = 0;
-      for i in 0..#chpl__primes.size {
-          if chpl__primes(i) > threshold {
-            prime = chpl__primes(i);
-            primeLoc = i;
-            break;
-          }
-      }
+    proc _findPowerOf2(numKeys:int) {
+      var n = (numKeys - 1): uint;
 
-      //No suitable prime found
-      if prime == 0 {
-        halt("Requested capacity (", numKeys, ") exceeds maximum size");
+      var k = 2;
+  
+      while n >> 1 > 0 {
+        n = n >> 1;
+        k = k << 1;
       }
-      return primeLoc;
+      return k << 1; // shift one more time so capacity is at less than half
```

Comment:
You can compute this with https://chapel-lang.org/docs/modules/standard/BitOps.html#BitOps.clz in fewer instructions. If you don't want to do that here you could add a TODO comment.

Reply:
Would you recommend doing this for all of the shifts or only this one? Trying it with all of them and I am getting some overflow, which I think makes sense since it is a rotation. Maybe use `rotl()` for all left shifts?

Reply:
@bmcdonald3 - you shouldn't be using `rotl` or `rotr` here. I was trying to get you to use `clz` because if you do then you can remove the `while` loop.

Reply:
OK, I see, thank you for clarifying. I think there may be a resolution error with using the `BitOps` module in the `ChapelHashtable` module though. Adding a `use BitOps` to this file causes a few tests to fail, for example `test/compflags/ferguson/print-module-resolution.chpl`, and I'm not sure exactly what this means, but I'm not sure if this is currently possible with the module resolution.

Reply:
Yes, let's:
 * put the shifts back (instead of rotates)
 * add a comment next to the while loop to say it could be removed and replaced with something using `ctz`
 * leave futher work on it for another PR